sfnpoller/api.go

```diff
@@ -52,6 +52,9 @@ func (a *API) BeginPolling(parentCtx context.Context) *API {
 func (a *API) Done() <-chan struct{} {
 	a.done = make(chan struct{})
 	go func() {
+		for _, task := range a.registeredTasks {
```

Comment:
Add a separate function shutDown() and move this code, too keep responsibility cleaner

Reply:
Added `Stop` func to decouple the responsibilities of `Done` func